### ISSUE TITLE
invalidated_update may cause a deadlock

### ISSUE THREAD

**Issue by M1ha-Shvn** [Suor/django-cacheops#367]
Hi.
Debugging one of my issues, I've found out that `invalidated_update(**kwargs)` method, called in transaction may cause race condition bug, leading to deadlock, as `objects = list(clone)` doesn't call select_for_update().
Example:
```python
with transaction.atomic():
    MyModel.objects.all().invalidated_update(param=1)
    MyModel.objects.all().update(param=1)
```
[Code, leading to deadlock](https://github.com/Suor/django-cacheops/blob/master/cacheops/query.py#L384)

My proposal is changing this part of code to:
```python
with transaction.atomic(using=clone.db):
    objects = list(clone.select_for_update())
    rows = clone.update(**kwargs)
    pks = {obj.pk for obj in objects}
    new_objects = self.model.objects.filter(pk__in=pks).using(clone.db)
```

**Comment by Suor** [comment]
Can you make a PR?

**Comment by M1ha-Shvn** [comment]
[Here](https://github.com/Suor/django-cacheops/pull/368) is it

**Comment by Suor** [comment]
Fixed in #368.
